plugins/outputs/postgresql/utils/utils.go

```diff
@@ -0,0 +1,158 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/jackc/pgx"
+)
+
+const (
+	insertIntoSQLTemplate = "INSERT INTO %s(%s) VALUES(%s)"
+)
+
+// GroupMetricsByMeasurement groups the list of metrics by the measurement name.
+// But the values are the index of the measure from the input list of measures.
+// [m, m, m2, m2, m] => {m:[0,1,4], m2:[2,3]}
+func GroupMetricsByMeasurement(m []telegraf.Metric) map[string][]int {
+	toReturn := make(map[string][]int)
+	for i, metric := range m {
+		var metricLocations []int
+		var ok bool
+		name := metric.Name()
+		if metricLocations, ok = toReturn[name]; !ok {
+			metricLocations = []int{}
+			toReturn[name] = metricLocations
+		}
+		toReturn[name] = append(metricLocations, i)
+	}
+	return toReturn
+}
+
+// BuildJsonb returns a byte array of the json representation
+// of the passed object.
+func BuildJsonb(data interface{}) ([]byte, error) {
+	d, err := json.Marshal(data)
+	if err != nil {
+		return nil, err
+	}
+	return d, nil
+}
+
+// QuoteIdent returns a sanitized string safe to use in SQL as an identifier
+func QuoteIdent(name string) string {
+	return pgx.Identifier{name}.Sanitize()
+}
+
+// QuoteLiteral returns a sanitized string safe to use in sql as a string literal
+func QuoteLiteral(name string) string {
+	return "'" + strings.Replace(name, "'", "''", -1) + "'"
```

Comment:
are there not helper functions in the pq or sql libs to do this for you? seems a bit risky to write our own sql sanitizers

Reply:
nothing found in the database/sql package but the pgx quoteIdentifier uses the same implementation for replacing " 
```
return `"` + strings.Replace(s, `"`, `""`, -1) + `"`
```

Reply:
Fair enough